src/utils/serial.cpp

```diff
@@ -1,20 +1,213 @@
 #include <string>
 
-#ifdef _WIN32
-#include <windows.h>
-#else
-#include <fcntl.h>
-#include <termios.h>
-#include <unistd.h>
+#include "serial.h"
+
+Serial::~Serial ()
+{
+}
+
+/////////////////////////////////////////////////
+/////////////////// LibFTDI /////////////////////
+/////////////////////////////////////////////////
+
+#ifdef USE_LIBFTDIPP
+
+// Note: LibFTDI also supports listing all connected devies.  This is not used yet.
+
+#include <chrono>
+#include <thread>
+
+#include <ftdi.hpp>
+#include <spdlog/spdlog.h>
+
+class LibFTDISerial : public Serial
+{
+public:
+    // "d:<devicenode>"
+    //      path of bus and device-node (e.g. "003/001") within usb device tree (usually at
+    //      /proc/bus/usb)
+    // "i:<vendor>:<product>"
+    //      first device with given vendor and product id, ids can be decimal, octal (preceded by
+    //      "0") or hex (preceded by "0x")
+    // "i:<vendor>:<product>:<index>"
+    //      as above with index being the number of the device (starting with 0) if there are more
+    //      than one
+    // "s:<vendor>:<product>:<serial>"
+    //      first device with given vendor id, product id and serial string
+    // there are also other ftdi constructors that take these items without parsing a string
+    LibFTDISerial (const char *description)
+        : description (description), port_open (false), logger (spdlog::get ("brainflow_logger"))
+    {
+    }
+
+    static bool is_libftdi (const char *port_name)
+    {
+        LibFTDISerial test (port_name);
+        return test.ctx.open (port_name) != -11;
+    }
+
+    void log_error (const char *action)
+    {
+        logger->error ("libftdi {}: {} -> {}", description, action, ctx.error_string ());
```

Comment:
I was actually trying to connect to board_logger here, see https://github.com/brainflow-dev/brainflow/pull/218/files#diff-707d50ff0c1f53ceed4b00bbb44de221f4e4804adeb6c8a62059677a2f153933R39 .  What is the right way to connect to board_logger ?  Maybe add a constructor parameter and pass it in?

Reply:
Its a public static field in a board class, I guess you can use forward declaration or include header(maybe it will lead to circular includes). 

Its preferred to use `safe_logger` instead spdlog directly, its a non static method of board class. It should be possible to pass it as an argument in constructor